enhancements/machine-config/manage-boot-images.md

```diff
@@ -0,0 +1,264 @@
+- `$(machine_set_name)-factory-image` storing the original provider image reference prior to when the feature was turned on
+- `$(machine_set_name)-factory-secret` storing the original stub secret name prior to the feature was turned on
+- `$(machine_set_name)-last-image` storing the last provider image reference before an update
+- `$(machine_set_name)-last-secret` storing the last stub secret name before an update
+
+##### Via new MachineSet fields
+
+Another proposed strategy is to add the required fields into the `.Status` section of a `MachineSet` object. This would be an API change.
+
+##### Mechanism
+
+A revert can be done by opting out the `MachineSet`, this will trigger the MSC to restore the MachineSet to last known good values stored from the backups. This is an important mitigation in case things go wrong(invalid bootimage references, incorrect patching... etc).
+
+The reason for keeping a factory version is in case the admin wishes to restore to factory values manually(they will have to opt-out of the feature first). It may also may aid in debugging.
+
+### Workflow Description
+
+It is important to note that there would be two "opt-in" knobs while this feature is under TechPreview. The admin would first have to turn on the feature gate, and then the opt-in mechanism. The secondary knob is necessary as some customers may want to keep their boot images static when this feature leaves TechPreview.
+Couple of strategies as before:
+
+##### via Annotations
+
+- To enroll a `MachineSet` for boot image updates, the cluster admin should add an annotation `io.openshift.mco-managed=true` to the `MachineSet`.
+- To un-enroll(and effectively revert) the `MachineSet` from boot image updates, the cluster admin should remove the `io.openshift.mco-managed=true` annotation from the `MachineSet`.
+
+##### via an Operator type 
+
+This is an API change to and would involve adding two new fields in the [operator types](https://github.com/openshift/api/blob/master/operator/v1/types_machineconfiguration.go) for the MCO:
+- `BootImageUpdateMode` This is an enum which can have three values: `Enabled`, `Selected` or `Disabled`
+- `BootImageUpdateEnrolledMachineSets` This is a list of enrolleds machinesets. When the above type is in the `Selected` mode, all machinesets in the list would be considered enrolled for updates.
+
+Here is a [rough PR](https://github.com/openshift/api/pull/1672) of what these API changes would look like.
+
+
+#### Variation and form factor considerations [optional]
+
+Any form factor using the MCO and `MachineSets` will be impacted by this proposal. So case by case:
+- Standalone OpenShift: Yes, this is the main target form factor.
+- microshift: No, as it does [not](https://github.com/openshift/microshift/blob/main/docs/contributor/enabled_apis.md) use `MachineSets`.
+- Hypershift: No, Hypershift does not have this issue.
+
+### API Extensions
+
+We may have to make some changes to MCO CRDs for the opt-in feature.
```

Comment:
This is where you should expand on the API extensions, include examples of the changes so that readers of the enhancement can see how they would interact with this feature. Either via go types or via YAML examples

Reply:
Oh - should I be describing the API changes/examples here then? (for the opt-in and backup mechanisms). I currently have them mentioned in the sections earlier.

Reply:
It would typically exist in the API extensions section, so I guess might be best to move here so that we are consistent with other EPs